lib/protocol/http2.js

```diff
@@ -0,0 +1,18 @@
+// TODO: remove after node@18 End-of-Life
+'use strict'
```

Comment:
Wouldnt it maybe faster if we do it like this?

```js
'use strict'

const status = { ':status': '103' }

function writeEarlyHints (reply, headers) {
  return new Promise(function (resolve) {
    if (reply.raw.socket === null) {
      resolve()
      return
    }
    Object.assign(headers, status)
    reply.raw.stream.additionalHeaders(headers)
    resolve()
  })
}

module.exports = writeEarlyHints
```

Reply:
It should resolve after the headers are added.
Not directly resolved.